sdks/java/core/src/main/java/org/apache/beam/sdk/util/MergingActiveWindowSet.java

```diff
@@ -144,7 +144,7 @@ public void ensureWindowIsActive(W window) {
     checkState(stateAddressWindows != null,
                              "Cannot ensure window %s is active since it is neither ACTIVE nor NEW",
                              window);
-    if (stateAddressWindows.isEmpty()) {
+    if (stateAddressWindows != null && stateAddressWindows.isEmpty()) {
```

Comment:
At least some of these are validations of the current state, i.e.:
```java
public void ensureWindowIsActive(W window) {
    Set<W> stateAddressWindows = activeWindowToStateAddressWindows.get(window);
    checkState(stateAddressWindows != null,
        "Cannot ensure window %s is active since it is neither ACTIVE nor NEW",
        window);
```

`IllegalStateException` seems like the right exception here. And looking forward, we shouldn't let FindBugs dictate which exceptions to throw.

Another possibility is to add a `@SuppressFBWarning` annotation for the line that is triggering the false-positive.